src/managed/OpenLiveWriter.BlogClient/Clients/WordPressClient.cs

```diff
@@ -53,5 +54,49 @@ protected override void ConfigureClientOptions(BlogClientOptions clientOptions)
             clientOptions.SupportsAuthor = true;
         }
 
+        public override BlogInfo[] GetUsersBlogs()
+        {
+            TransientCredentials tc = Login();
+            return GetWPUsersBlogs(tc.Username, tc.Password);
+        }
+        private BlogInfo[] GetWPUsersBlogs(string username, string password)
+        {
+            // call method
+            XmlNode result = CallMethod("wp.getUsersBlogs",
+                new XmlRpcString(username),
+                new XmlRpcString(password, true));
+
+            try
+            {
+                // parse results
+                ArrayList blogs = new ArrayList();
+                XmlNodeList dataValues = result.SelectNodes("array/data");
+                foreach (XmlNode dataValue in dataValues)
+                {
+                    XmlNodeList blogNodes = dataValue.SelectNodes("value/struct");
+                    foreach (XmlNode blogNode in blogNodes)
+                    {
+                        // get node values
+                        XmlNode idNode = blogNode.SelectSingleNode("member[name='blogid']/value");
+                        XmlNode nameNode = blogNode.SelectSingleNode("member[name='blogName']/value");
+                        XmlNode urlNode = blogNode.SelectSingleNode("member[name='url']/value");
+
+                        // add to our list of blogs
+                        blogs.Add(new BlogInfo(idNode.InnerText, HttpUtility.HtmlDecode(NodeToText(nameNode)), urlNode.InnerText));
+                    }
+
+                    // return list of blogs
+                    return (BlogInfo[])blogs.ToArray(typeof(BlogInfo));
```

Comment:
You have a `return` in a `foreach`, which doesn't seem right. Looking at the base class implementation in `BloggerCompatibleClient.GetUsersBlogs()`, why not just use the same looping strategy from there? Just call `XmlNodeList blogNodes = result.SelectNodes("array/data/value/struct");` instead of having two `foreach` loops?